src/net_builtin.c

```diff
@@ -552,17 +554,24 @@ long mg_io_send(struct mg_connection *c, const void *buf, size_t len) {
   struct mg_tcpip_if *ifp = (struct mg_tcpip_if *) c->mgr->priv;
   struct connstate *s = (struct connstate *) (c + 1);
   uint32_t rem_ip;
+  size_t eth_h_len = 14, ip_max_h_len = 24, tcp_max_h_len = 60, udp_h_len = 8;
+  size_t max_headers_len = eth_h_len + ip_max_h_len +
+                          (c->is_udp ? udp_h_len : tcp_max_h_len);
   memcpy(&rem_ip, c->rem.ip, sizeof(uint32_t));
+  if (len + max_headers_len > ifp->tx.len) {
+    len = ifp->tx.len - max_headers_len;
+  }
+  if (s->mtu < max_headers_len - eth_h_len) {
+    s->mtu = MIP_MTU_DEFAULT;
+  }
+  if (len + max_headers_len - eth_h_len > s->mtu) {
+    len = s->mtu - max_headers_len + eth_h_len;
+    if (c->is_udp)
+      MG_ERROR(("UDP datagram exceeds MTU"));
```

Comment:
Please tell the datagram has been truncated, something like
"Truncated UDP datagram exceeding MTU"